### PR TITLE
Removing life and use score

### PR DESCRIPTION
**Pull request by bronhaim**
Signed-off-by: Yaniv Bronhaim <ybronhei@redhat.com>

### CODE REVIEW

File: rose/server/score_test.py [hunk 1]
```diff
@@ -87,10 +87,10 @@ class TestPenguin(SinglePlayerTest):
     def test_pickup(self):
         self.player.action = actions.PICKUP
         self.process()
-        # Player move up and get more life
+        # Player move up and get more score
         assert self.player.x == self.x
         assert self.player.y == self.y - 1
-        assert self.player.life == self.life + 1
+        assert self.player.score == self.score + 1
```
Comment (nirs) [inline rose/server/score_test.py]: Shouldn't this be self.score + config.score_penguin_catch?

File: rose/server/score_test.py [hunk 2]
```diff
@@ -227,14 +227,14 @@ def test_forward(self):
         self.obstacle = obstacles.PENGUIN
         self.track.set(self.x, self.y, self.obstacle)
         self.process()
-        # Player keep position but get more life
+        # Player keep position but get more score
         assert self.player.x == self.x
         assert self.player.y == self.y
-        assert self.player.life == self.life + 1
+        assert self.player.score == self.score + 1
```
Comment (nirs) [inline rose/server/score_test.py]: should use config, not hardcoded value. Otherwise we are testing the config instead of the logic.

File: rose/server/score_test.py [hunk 3]
```diff
@@ -268,85 +268,85 @@ def test_player_in_lane_wins(self):
...
+        self.player2.score = 0
         self.player2.action = actions.PICKUP
         self.process()
         # Player 1 win because it is in lane
         assert self.player1.x == 1
         assert self.player1.y == 5
-        assert self.player1.life == 0
-        # Player 2 got more life but move back
+        assert self.player1.score == 0
+        # Player 2 got more score but move back
```
Comment (nirs) [inline rose/server/score_test.py]: Don't we add score now in these cases?

**Comment by bronhaim** [comment]
Dont take it yet! just saw that its still buggy
the car can cross to other side

File: rose/server/score.py [hunk 4]
```diff
@@ -14,12 +14,19 @@ def process(players, track):
...
                 player.x -= 1
+	    else:
+		player.score -= config.move_backward
+
         elif player.action == actions.RIGHT:
-            if player.x < config.matrix_width - 1:
+            if player.x < player_start_point + config.cells_per_player - 1:
                 player.x += 1
+	    else:
+		player.score -= config.move_backward
```
Comment (nirs) [inline rose/server/score.py]: Please separate this to another patch, this is not related to replacing life with score.

Are you trying to prevent player from moving to the other lane? I don't think we want this behavior.

File: rose/server/score.py [hunk 5]
```diff
@@ -36,7 +42,8 @@ def process(players, track):
                 player.y += 1
                 player.score += config.move_backward
             else:
-                player.score += config.jump
+                player.score += config.jump + config.move_forward
```
Comment (nirs) [inline rose/server/score.py]: Not related to replacing life with score.

File: rose/server/score.py [hunk 6]
```diff
@@ -49,14 +56,14 @@ def process(players, track):
                 player.y += 1
                 player.score += config.move_backward
             else:
-                player.score += config.brake
+                player.score += config.brake + config.move_forward
```
Comment (nirs) [inline rose/server/score.py]: Not related

File: rose/server/score.py [hunk 7]
```diff
@@ -49,14 +56,14 @@ def process(players, track):
...
         elif obstacle == obstacles.PENGUIN:
             if player.action == actions.PICKUP:
                 track.clear(player.x, player.y)
                 player.y -= 1
-                player.life += 1
-                player.score += config.move_forward
+                player.score += config.score_penguin_catch + config.move_forward
```
Comment (nirs) [inline rose/server/score.py]: Not related

File: rose/server/score.py [hunk 8]
```diff
@@ -49,14 +56,14 @@ def process(players, track):
...
         elif obstacle == obstacles.NONE:
+            player.score += config.move_forward
```
Comment (nirs) [inline rose/server/score.py]: Not related

File: rose/server/score_test.py [hunk 9]
```diff
@@ -87,10 +87,10 @@ class TestPenguin(SinglePlayerTest):
     def test_pickup(self):
         self.player.action = actions.PICKUP
         self.process()
-        # Player move up and get more life
+        # Player move up and get more score
         assert self.player.x == self.x
         assert self.player.y == self.y - 1
-        assert self.player.life == self.life + 1
+        assert self.player.score == self.score + config.score_penguin_catch + config.move_forward
```
Comment (nirs) [inline rose/server/score_test.py]: not related (adding both pick and move forward)

Comment (bronhaim) [inline rose/server/score_test.py]: of course it is.. this is how it works currently

File: rose/server/score.py (hunk 4 above)
Comment (bronhaim) [inline rose/server/score.py]: I Don't agree. we have to have it.. maybe not to take points in that case. but with this formula at least you know if the user moved to the other side

File: rose/server/score.py [hunk 10]
```diff
@@ -14,11 +14,13 @@ def process(players, track):
...
+        player_start_point = player.lane * config.cells_per_player
         if player.action == actions.LEFT:
-            if player.x > 0:
+            # the left player should be start on x=0, and right in the middle
+            if player.x > player_start_point:
                 player.x -= 1
         elif player.action == actions.RIGHT:
-            if player.x < config.matrix_width - 1:
+            if player.x < player_start_point + config.cells_per_player - 1:
                 player.x += 1
```
Comment (nirs) [inline rose/server/score.py]: Not related to replacing life with score. Please do this in another patch and explain what are you trying to do.

File: rose/server/score_test.py (hunk 9 above)
Comment (nirs) [inline rose/server/score_test.py]: Looks like it works like that because you changed it in this patch. If you remove everything which is not related to replacing life with score, will this test be the same?

File: rose/server/score.py (hunk 10 above)
Comment (bronhaim) [inline rose/server/score.py]: i intend to disagree. this is the logic to have it works. without it the score means nothing

File: rose/server/score.py (hunk 5 above)
Comment (bronhaim) [inline rose/server/score.py]: wahh?! I really don't get you man

File: rose/server/score_test.py (hunk 9 above)
Comment (bronhaim) [inline rose/server/score_test.py]: no...

Comment (bronhaim) [inline rose/server/score_test.py]: the commit that already in changed the way the score is counted..
